python/grass/temporal/register.py

```diff
@@ -226,7 +226,9 @@ def register_maps_in_space_time_dataset(
                 found = gscript.find_file(element=type, name=mapname)
                 if found["mapset"] is not None and len(found["mapset"]) > 0:
                     map_mapset = found["mapset"]
-            row["id"] = AbstractMapDataset.build_id(mapname, map_mapset)
+                row["id"] = AbstractMapDataset.build_id(mapname, map_mapset)
+            else:
+                row["id"] = AbstractMapDataset.build_id(*mapname.split("@")[0:2])
```

Comment:
We had a long discussion how to do this right. Perhaps a library function would be the best solution to cover the split or for the whole procedure in this block of code.

Reply:
Here, I meant a function to do the splitting of map@mapset (name@subproject), but the case here is more specific given the layer number is involved too. Also the error handling (fatal versus raise) and existence requirement might be different here than other places (existing vs new).

Reply:
Using the new `build_id_from_search_path` function now